lib/phoenix/live_dashboard/live/home_live.ex

```diff
@@ -172,59 +178,29 @@ defmodule Phoenix.LiveDashboard.HomeLive do
 
         <div class="card mb-4">
           <div class="card-body resource-usage">
-
-            <div class="progress flex-grow-1 mb-3">
-              <%= for {_, section_name, section_value, color} <- memory_usage_sections(@system_usage.memory) do %>
-                <div
-                  title="<%=section_name %> - <%=percentage(section_value, @system_usage.memory.total, round: true) %>%"
-                  class="progress-bar bg-gradient-<%= color %>"
-                  role="progressbar"
-                  aria-valuenow="<%=section_value %>"
-                  aria-valuemin="0"
-                  aria-valuemax="<%=@system_usage.memory.total %>"
-                  style="width: <%=percentage(section_value, @system_usage.memory.total) %>%">
-                </div>
-              <% end %>
-            </div>
-
-            <div class="resource-usage-legend">
-
-              <div class="resource-usage-legend-entries row flex-column flex-wrap">
-                <%= for {_, section_name, section_value, color} <- memory_usage_sections(@system_usage.memory) do %>
-                  <div class="col-lg-6 resource-usage-legend-entry d-flex align-items-center py-1 flex-grow-0">
-                    <div class="resource-usage-legend-color bg-<%= color %> mr-2"></div>
-                    <span><%=section_name %></span>
-                    <span class="flex-grow-1 text-right text-muted">
-                      <%= format_bytes(section_value) %>
-                    </span>
-                  </div>
-                <% end %>
-              </div>
-
-              <div class="row">
-                <div class="col">
-                  <div class="resource-usage-total text-center py-1 mt-3">
-                    Total usage: <%= format_bytes(@system_usage.memory[:total]) %>
-                  </div>
+            <%= live_component @socket, ColorBarComponent, id: :usage, data: memory_usage_sections_percent(@system_usage.memory, @system_usage.memory.total) %>
+            <%= live_component @socket, ColorBarLegendComponent, id: :usage_legend, data: memory_usage_sections(@system_usage.memory), fn_format: &format_bytes(&1) %>
```

Comment:
I believe we don't need ID. Those components can be stateless.

Reply:
Also, maybe we should merge the ColorBar and ColorBarLegend into once. At the moment, do we plan to use them separately anywhere?

Reply:
currently I have 2 cpu modules - and 1 for memory usage (plus this one from the home page)
![image](https://user-images.githubusercontent.com/904179/80407998-6a7d9400-88be-11ea-98d2-59c7a04edd95.png) this can be merged back into one - it was just easier to work in two

Reply:
I also use different formatting in memory - % on bar and bytes on the legend

Reply:
I see, sounds good then. Let's just remove the `ID` then and let's rename `fn_format` to formatter. Then we can ship it!